prac_05/hex_colours.py

```diff
@@ -1,16 +1,20 @@
 """CP1404 Practical 5 - Hex Colours"""
 
-"""Based on the state name example program above, create a program that allows 
-you to look up hexadecimal colour codes like those at http://www.color-hex.com/color-names.html
-
-Use a constant dictionary of about 10 colour names and write a program that allows a user 
-to enter a name and get the code, e.g., entering AliceBlue (or aliceblue - make it case-independent) should show #f0f8ff.
-
-Entering an invalid colour name should not crash the program.
-Allow the user to enter names until they enter a blank one to stop the loop.
-
-Note
-We have just done two exercises that use dictionaries that are constants and named in ALL_CAPS.
-Please don't think this is any kind of rule or pattern.
-Dictionaries that change would not be constants."""
+COLOUR_CODES = {"aliceblue": "#f0f8ff",
+                "antiquewhite": "#faebd7",
+                "aqua": "#00ffff",
+                "aquamarine": "#7fffd4",
+                "azure": "#f0ffff",
+                "beige": "#f5f5dc",
+                "bisque": "#ffe4c4",
+                "black": "#000000",
+                "blue": "#0000ff",
+                "coral": "#ff7f50"}
 
+colour_name = input("Enter a colour name: ").lower()
+while colour_name != "":
+    try:
+        print(f"The code for \"{colour_name}\" is {COLOUR_CODES.get(colour_name)}")
+    except KeyError:
+        print("Invalid short state")
```

Comment:
I like how you have written in this except in this program, however in the print statement perhaps you could use a different name from short state?